compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -1635,7 +1645,18 @@ public void visit(BLangWhile astWhileStmt) {
         astWhileStmt.body.accept(this);
         this.env.unlockVars.pop();
         if (this.env.enclBB.terminator == null) {
-            this.env.enclBB.terminator = new BIRTerminator.GOTO(null, whileExprBB, this.currentScope);
+            if (astWhileStmt.body.pos != null) {
```

Comment:
Can we get this null?

Reply:
We can remove the null check since the position never takes null here. Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/32667/commits/d6c32a58981fff76246a855c20aef34596fead4f